src/main/java/slaybot/Parser.java

```diff
@@ -0,0 +1,176 @@
+package slaybot;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import entity.Deadline;
+import entity.Event;
+import entity.ToDo;
+import exception.InvalidDeadlineException;
+import exception.InvalidEventException;
+import exception.InvalidTodoException;
+import exception.UnknownCommandException;
+/**
+ * The Parser class is responsible for parsing user input and converting it
+ * into meaningful objects or commands for the SlayBot application. It handles the
+ * parsing of commands related to tasks (ToDo, Deadline, Event) and other user interactions.
+ */
+public class Parser {
+
+    enum Command {
```

Comment:
Good use of enums here to represent commands, something that I can learn from you :)